src/main/scala/bubblewrap/HttpClient.scala

```diff
@@ -1,27 +1,24 @@
 package bubblewrap
 
 import java.net.URL
-import java.util.concurrent.TimeUnit
+import java.util.concurrent.{Executors, TimeUnit}
 
 import io.netty.handler.codec.http.HttpHeaders.Names._
 import io.netty.handler.codec.http.HttpHeaders.Values._
 import io.netty.handler.codec.http.cookie.ClientCookieDecoder
 import io.netty.handler.ssl.util.InsecureTrustManagerFactory
 import io.netty.handler.ssl.{SslContextBuilder, SslProvider}
 import io.netty.util.HashedWheelTimer
-import org.asynchttpclient.netty.NettyResponseFuture
-import org.asynchttpclient.netty.channel.DefaultChannelPool
 import org.asynchttpclient.proxy.ProxyServer
 import org.asynchttpclient.{DefaultAsyncHttpClient, DefaultAsyncHttpClientConfig, Realm}
 
 import scala.collection.JavaConverters._
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.util.Try
+import scala.concurrent.{ExecutionContext, Promise}
 
 class HttpClient(clientSettings: ClientSettings = ClientSettings()) {
   val lenientSSLContext = SslContextBuilder.forClient().sslProvider(SslProvider.JDK).trustManager(InsecureTrustManagerFactory.INSTANCE).build()
 
-  val timer = new HashedWheelTimer(100, TimeUnit.MILLISECONDS, 3072)
+  val timer = new HashedWheelTimer(100, TimeUnit.MILLISECONDS, 100)
```

Comment:
Some notes:

1. Time to visit all the slots - old: 5 mins (3000*100ms = 300 secs = 5 mins). In the new one - (100*100 = 10 seconds)
2. Each slot contains a list of `TimeoutTasks`. Now, since the # of slots have reduced, we might face issues with more items in each slot - creating overhead during timeout enqueue operation + during timeout firing operation. In short, the # of slots directly proportional to concurrency.

We might have to check this up to see if we are spending a lot of time on it and tweak based on it.

Reply:
@brewkode Agreed. Its tradeoff for which i don't have an answer for. I think we can be aggressive and then tone it down so that we have a good balance between resources vs queue ops.